var/spack/repos/builtin/packages/arm/package.py

```diff
@@ -19,18 +67,36 @@ class Arm(Package):
     homepage = "https://developer.arm.com/tools-and-software/server-and-hpc/arm-allinea-studio"
     url = "https://developer.arm.com/-/media/Files/downloads/hpc/arm-allinea-studio/20-2-1/Ubuntu16.04/arm-compiler-for-linux_20.2.1_Ubuntu-16.04_aarch64.tar"
 
-    # FIXME: The version is checksummed for Ubuntu 16.04, but this is not
-    # FIXME: important at the moment since the package is only meant to
-    # FIXME: provide detection
-    version('20.2.1', sha256='dc3f945b05b867809d9b507cb8ebba9cf72a6818d349207dbe1392c13dc0ad79')
+    maintainers = ['OliverPerks']
 
-    def install(self, spec, prefix):
-        raise InstallError(
-            'No install method available yet, only system detection.'
-        )
+    # Build Versions: establish OS for URL
+    acfl_os = get_os()
+
+    # Build Versions
+    for ver, packages in _versions.items():
+        pkg = packages.get(acfl_os)
+        if pkg:
+            version(ver, sha256=pkg[0], url=pkg[1])
+
+    # Only install for Aarch64
+    conflicts('target=x86_64:', msg='Only available on Aarch64')
+    conflicts('target=ppc64:', msg='Only available on Aarch64')
+    conflicts('target=ppc64le:', msg='Only available on Aarch64')
 
     executables = [r'armclang', r'armclang\+\+', r'armflang']
 
+    # Licensing
+    license_required = True
+    license_comment = "#"
+    license_files = ["licences/Licence"]
+    license_vars = ["ARM_LICENSE_DIR"]
+    license_url = "https://developer.arm.com/tools-and-software/server-and-hpc/help/help-and-tutorials/system-administration/licensing/arm-licence-server"
+
+    # Run the installer with the desired install directory
+    def install(self, spec, prefix):
+        exe = './arm-compiler-for-linux_{0}_{1}.sh'.format(spec.version, get_os())
+        subprocess.call([exe, "--accept", "--force", "--install-to", prefix])
```

Comment:
Here you can use `Executable` (Spack's wrapper around subprocess calls) instead of `subprocess.call` directly. I didn't test the code, but it should be similar to:
```python
exe = Executable('./arm-compiler-for-linux_{0}_{1}.sh'.format(spec.version, get_os()))
exe("--accept", "--force", "--install-to", prefix)
```